hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop-titan-provider/src/main/java/org/hawkular/inventory/impl/tinkerpop/provider/TitanProvider.java

```diff
@@ -248,4 +249,17 @@ public RuntimeException translateException(RuntimeException inputException, Cano
         }
         return inputException;
     }
+
+    @Override public boolean requiresRollbackAfterFailure(Throwable t) {
+        while (t != null) {
+            if (t instanceof TitanException
+                    && "Could not commit transaction due to exception during persistence".equals(t.getMessage())) {
+                return false;
+            }
+
+            t = t.getCause();
+        }
```

Comment:
FYI https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Throwables.html#getRootCause(java.lang.Throwable)

But I guess, guava is not on the classpath, or is it?